libceed: Add BLAS_DIR and link time blas dependency

cameronrutherford opened the pull request:
I found this was necessary, or else libceed would complain about `-lblas` not being found.

The alternative workaround was to `spack load blas` before building libceed, which isn't ideal. Perhaps this change is wrong though, and I was doing something wrong in building.

cameronrutherford committed:
libceed: Add BLAS_DIR and link time blas dependency

cameronrutherford commented:
@spackbot style --fix

cameronrutherford committed:
Fix spack style.

jedbrown commented:
What issue are you encountering? libCEED does not call BLAS directly, but libXSMM has a transitive dependency on BLAS so if you enable libXSMM, you should also specify `BLAS_LIB` (because spack strives to be explicit). If that's consistent with what you've observed, can you update this PR to only be used when libXSMM is enabled?

cameronrutherford committed:
libceed: update blas dep to be based on ~/+libxsmm

cameronrutherford commented:
@jedbrown thanks for the review. I updated based on your suggestion.

I should've documented the error, but to the best of my memory, it was that if you don't have a system blas, there is a link call to `-lblas` inserted when spack was building libceed. [This commit](https://github.com/awslabs/palace/pull/362/commits/3d33c1331756976e11c085e1fa170aa8ef0c0457) is where I implemented this change, and you can see that I ended up being able to remove the call to `spack load intel-oneapi-mkl` that I hacked in while I was developing.

jedbrown reviewed (approved):
This looks correct to me, thanks!
